api/core/v1alpha1/vtep_types.go

```diff
@@ -0,0 +1,163 @@
+	// +required
+	PrimaryInterfaceRef LocalObjectReference `json:"primaryInterfaceRef"`
+
+	// AnycastInterfaceRef is the reference to the loopback interface used for anycast VTEP IP address.
+	// +required
+	AnycastInterfaceRef LocalObjectReference `json:"anycastInterfaceRef,omitempty"`
+
+	// SuppressARP indicates whether ARP suppression is enabled for this VTEP.
+	// +required
+	SuppressARP bool `json:"suppressARP"`
+
+	// HostReachability specifies the method used for host reachability.
+	// +required
+	// +kubebuilder:validation:Enum=FloodAndLearn;BGP
+	HostReachability HostReachabilityType `json:"hostReachability"`
+
+	// MulticastGroup specifies the multicast group configuration for the VTEP.
+	// +optional
+	MulticastGroup *MulticastGroup `json:"multicastGroup,omitempty"`
+}
+
+// HostReachabilityType defines the method used for host reachability.
+type HostReachabilityType string
+
+const (
+	HostReachabilityTypeBGP           HostReachabilityType = "BGP"
+	HostReachabilityTypeFloodAndLearn HostReachabilityType = "FloodAndLearn"
+)
+
+type MulticastGroup struct {
+	// Type specifies the multicast group level (L2 or L3).
+	// +required
+	// +kubebuilder:validation:Enum=L2;L3
+	Type MulticastGroupType `json:"type"`
+
+	// Prefix is the multicast group IP prefix.
+	// +required
+	Prefix IPPrefix `json:"prefix"`
+}
+
+// MulticastGroupType defines the type of multicast group (L2 or L3).
+type MulticastGroupType string
+
+const (
+	MulticastGroupTypeL2 MulticastGroupType = "L2"
+	MulticastGroupTypeL3 MulticastGroupType = "L3"
+)
+
+// VTEPStatus defines the observed state of VTEP.
+type VTEPStatus struct {
+	// For Kubernetes API conventions, see:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties
+
+	// conditions represent the current state of the VTEP resource.
+	// Each condition has a unique type and reflects the status of a specific aspect of the resource.
+	//
+	// Standard condition types include:
+	// - "Available": the resource is fully functional
+	// - "Progressing": the resource is being created or updated
+	// - "Degraded": the resource failed to reach or maintain its desired state
+	//
+	// The conditions are a list of status objects that describe the state of the VTEP.
+	//+listType=map
+	//+listMapKey=type
+	//+patchStrategy=merge
+	//+patchMergeKey=type
+	//+optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=vteps
+// +kubebuilder:resource:singular=vtep
+// +kubebuilder:printcolumn:name="VTEP",type=string,JSONPath=`.spec.name`
+// +kubebuilder:printcolumn:name="Device",type=string,JSONPath=`.spec.deviceRef.name`
+// +kubebuilder:printcolumn:name="Ready",type=string,JSONPath=`.status.conditions[?(@.type=="Ready")].status`
```

Comment:
Can we add Source Interface, Anycast Source Interface and Host Reachability as columns?